lib/Dialect/Ibis/Transforms/IbisPortrefLowering.cpp

```diff
@@ -0,0 +1,340 @@
+//===- IbisPortrefLowering.cpp - Implementation of PortrefLowering --------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "PassDetails.h"
+
+#include "circt/Dialect/Ibis/IbisDialect.h"
+#include "circt/Dialect/Ibis/IbisOps.h"
+#include "circt/Dialect/Ibis/IbisPasses.h"
+#include "circt/Dialect/Ibis/IbisTypes.h"
+
+#include "mlir/IR/Builders.h"
+#include "mlir/Transforms/DialectConversion.h"
+#include "llvm/ADT/TypeSwitch.h"
+
+using namespace mlir;
+using namespace circt;
+using namespace ibis;
+
+namespace {
+
+template <typename TPort>
+class PortLoweringPattern : public OpConversionPattern<TPort> {
```

Comment:
What's the point of having an empty parent class? I assume it was intended for code sharing...?

Reply:
Yup, leftover code. Will remove.